plenum/server/replica.py

```diff
@@ -526,6 +526,20 @@ def on_view_change_done(self):
             raise LogicError("{} is not a master".format(self))
         self.last_prepared_before_view_change = None
 
+    def clear_requests_and_fix_last_ordered(self):
+        reqs_for_remove = []
+        for key in self.requests:
+            ledger_id, seq_no = self.node.seqNoDB.get(key)
+            if seq_no is not None:
+                reqs_for_remove.append((key, ledger_id, seq_no))
+        for key, ledger_id, seq_no in reqs_for_remove:
+            self.requests.free(key)
+            self.requestQueues[int(ledger_id)].discard(key)
+        master_last_ordered_3pc = self.node.master_replica.last_ordered_3pc
+        if master_last_ordered_3pc[0] >= self.last_ordered_3pc[0] \
+                and master_last_ordered_3pc[1] > self.last_ordered_3pc[1]:
+            self.last_ordered_3pc = master_last_ordered_3pc
```

Comment:
We store here the value of the tuple, without the risk of passing the reference, is it?